actgen/nnutils.py

```diff
@@ -95,23 +95,23 @@ def soft_copy_from(self, other, tau=0.1):
 
 class MLP(Network):
     def __init__(self, n_inputs, n_outputs, n_hidden_layers, n_units_per_layer, 
-                use_dropout=False, dropout_rate=0.5):
+                dropout=False):
         super().__init__()
         self.n_inputs = n_inputs
         self.n_outputs = n_outputs
 
         assert n_hidden_layers >= 1
         
-        if use_dropout:
+        if dropout:
```

Comment:
This section can be simplified by using a lambda `add_dropout_if_enabled(dropout)` that returns either a `torch.nn.Dropout` or a `torch.nn.Identity` layer

Reply:
does `torch.nn.Identity` make the network to do extra work? Because I assume most of the time we will have `dropout = False`

Reply:
I think the extra work is negligible, but we could also just avoid the Identity like this:

```python
layers = [
    # [...]
    torch.nn.ReLU(),
] + ([torch.nn.Dropout(dropout)] if dropout else []) + [
    # [...]
]
```